Adding ESR docs (#1240)
balena-io/docs: pages/reference/OS/extended-support-release.md

---
title: Extended Support Release Process
excerpt: An overview of the {{ $names.company.lower }} extended support release process
---
# Extended Support Release (ESR) process

__Note:__ This feature is only available on [Production and Enterprise plans][billing].

The Extended Support Release (ESR) process allows fleet owners to update the host OS on their devices at most twice a year to ensure they are on a supported version. New ESR versions are released on a three-month schedule, and as such, fleet owners know in advance when releases will be available and when they need to upgrade.

Each ESR release is supported for nine months from the time of release. A supported ESR version implies that high-risk security vulnerabilities and critical bug fixes will be backported to that version. Backports of any functional enhancements are not in scope.

## ESR schedule

Three lines of ESR versions are supported, known as _next_, _current_, and _sunset_. Every three months, a new ESR version is released, and each line will transition forward at that time as illustrated on the following diagram:

![ESR schedule](/img/common/esr-process.png)

> For example, an ESR version is released known as `2019.10`, which becomes the _next-ESR_. In three months, a new ESR version, i.e., `2020.04`, becomes the _next-ESR_ line, while `2019.10` transitions to become the _current-ESR_. Similarly, in a further three months, there is a new ESR version, and `2019.10` becomes the _sunset-ESR_. In a further three months, and nine months after release, the `2019.10` ESR version will reach its end-of-life, and no further updates will be offered, and users should update to one of the supported releases to receive future updates by performing a [self-service update][self-service-updates].

Reviewer: ```suggestion
For example, an ESR version is released known as `2019.10`, which becomes the _next-ESR_. In three months, a new ESR version, i.e., `2020.04`, becomes the _next-ESR_ line, while `2019.10` transitions to become the _current-ESR_. Similarly, three months after that, there is a new ESR version, and `2019.10` becomes the _sunset-ESR_. Three months later and nine months after release, the `2019.10` ESR version will reach its end-of-life, and no further updates will be offered, and users should update to one of the supported releases to receive future updates by performing a [self-service update][self-service-updates].
```

> For example, an ESR version is released known as `2019.10`, which becomes the _next-ESR_. In three months, a new ESR version, i.e., `2020.04`, becomes the _next-ESR_ line, while `2019.10` transitions to become the _current-ESR_. Similarly, three months after that, there is a new ESR version, and `2019.10` becomes the _sunset-ESR_. Three months later and nine months after release, the `2019.10` ESR version will reach its end-of-life, and no further updates will be offered, and users should update to one of the supported releases to receive future updates by performing a [self-service update][self-service-updates].

Reviewer 2: Oooh, just spotted another small problem :wink: 
Wouldn't three months after `2019.10` be `2020.01`, rather than `2020.04` ?

Author: Thanks again, really appreciate it and that's fixed. I think I've rewritten this one too many times 😑

> __Note:__ ESR versions are named by year and month `yyyy.mm` and contain a patch version number. So, `2019.10.4` would be the fifth release of the `2019.10` ESR version.

Reviewer 2: I wonder if it's worth clarifying this by adding something like " (`2019.10.0` would be the first release)" at the end? :man_shrugging:

> __Note:__ ESR versions are named by year and month `yyyy.mm` and contain a patch version number. So, `2019.10.4` would be the fourth release of the `2019.10` ESR version.

Reviewer 2: Is there ever a 'bare' `2019.10` version, or will the first version always be e.g. `2019.10.1` ? (i.e. is there _always_ a patch version number?)

Author: It'll start from `2019.10.0` so yes always a patch number.

Reviewer 2: If it starts at `2019.10.0`, wouldn't that make `2019.10.4` the _fifth_ release? :confused:

Author: It would, nice catch 😄Let me double-check the 0-indexed patch version before I update this.

Author: Thanks again, confirmed it's 0-indexed and fixed that error you highlighted via the latest commit.

## Using an ESR host OS version

### Adding a new device

> For new devices, if you are on a Production or Enterprise plan with a [supported device type](#supported-devices), when you add a new device, you will be given the option to _Select OS type_ which defaults to _esr_. If you would like a non-ESR version, expand this dropdown and select _default_ for the host OS type.

Reviewer 3: There were some changes here, so the options would be `default` -> `balenaOS`, `esr` -> `balenaOS ESR`.
Also, the `Select OS type` is visible for everyone, but `balenaOS ESR` will be disabled for non-production or non-enterprise users. It will also be disabled if we currently don't have any versions for ESR (when a device type is not supported yet).

Author: Updated and included a new screenshot. I didn't address the non-supported devices comment as the paragraph is predicated on the fact the user has a supported device type. I tested this and it's also very intuitive.

<img src="/img/common/esr-new-device.png" alt="Add new ESR device" width="80%">

Next, select the ESR version as either _next_, _current_, or _sunset_ if available. The _next_ version is selected by default and offers at least six months (and up to nine months) of critical backports and fixes.

> ### hostOS update

Reviewer: ```suggestion
### Host OS update
```

for consistency

Author: I'm so making a list of all of these today... 😄

> For those users on a Production or Enterprise plan with an existing [supported device](#supported-devices), you can update to an ESR version via a [self-serve update][self-service-updates]. You should select the _esr_ host OS type and your chosen ESR version.

Reviewer 3: Same with comment above.

__Note:__ Once updated to an ESR version, it is not possible to update from an ESR host OS version to a non-ESR one.

## Supported devices

> ESR hostOS versions are currently only available for the following devices:

Reviewer: ```suggestion
ESR host OS versions are currently only available for the following devices:
```

just for consistency

* Raspberry Pi (v1 and Zero)

> * Raspberry Pi 3

Reviewer 3: Raspberry Pi 4 is also in the pipeline, and it should be available very soon. I would say adding it right away in here.

Each ESR release is supported for nine months from the time of release. A supported ESR version implies that high-risk security vulnerabilities and critical bug fixes will be backported to that version. Backports of any functional enhancements are not in scope.

## ESR schedule

Three lines of ESR versions are supported, known as _next_, _current_, and _sunset_. Every three months, a new ESR version is released, and each line will transition forward at that time as illustrated on the following diagram:

![ESR schedule](/img/common/esr-process.png)

For example, an ESR version is released known as `2019.10`, which becomes the _next-ESR_. In three months, a new ESR version, i.e., `2020.01`, becomes the _next-ESR_ line, while `2019.10` transitions to become the _current-ESR_. Similarly, three months after that, there is a new ESR version, and `2019.10` becomes the _sunset-ESR_. Three months later and nine months after release, the `2019.10` ESR version will reach its end-of-life, and no further updates will be offered, and users should update to one of the supported releases to receive future updates by performing a [self-service update][self-service-updates].

> __Note:__ ESR versions are named by year and month `yyyy.mm` and contain a patch version number starting at 0. So, `2019.10.4` would be the fifth release of the `2019.10` ESR version.

Reviewer 4: I would maybe move this before the example. We could also add something like: 

Since ESR versions will *only* receive backports for high-risk security vulnerabilities and critical bug fixes, new releases will only increment the patch number, this means the major and minor will remain fixed during the lifecycle of an ESR version and only the patch number will keep increasing as new versions are released.

Author: Thanks. I moved this and with the additional content, it was a bit long for a note. I've added some clarifying text that only patch number will increase during the lifecycle of an ESR version while minimizing the amount of text. If you still think anything isn't clear just let me know :)

## Using an ESR host OS version

### Adding a new device

For new devices, if you are on a Production or Enterprise plan with a [supported device type](#supported-devices), when you add a new device, you will be given the option to _Select OS type_ which defaults to _{{$names.os.lower}} ESR_. If you would like a non-ESR version, expand this dropdown and select _{{$names.os.lower}}_ for the host OS type.

<img src="/img/common/esr-new-device.png" alt="Add new ESR device" width="80%">

Next, select the ESR version as either _next_, _current_, or _sunset_ if available. The _next_ version is selected by default and offers at least six months (and up to nine months) of critical backports and fixes.

### Host OS update

For those users on a Production or Enterprise plan with an existing [supported device](#supported-devices), you can update to an ESR version via a [self-serve update][self-service-updates]. You should select the _{{$names.os.lower}} ESR_ host OS type and your chosen ESR version.

__Note:__ Once updated to an ESR version, it is not possible to update from an ESR host OS version to a non-ESR one.

## Supported devices

> ESR host OS versions are currently only available for the following devices:

Reviewer 4: Maybe add:

but we plan to keep adding devices to this list as we identify which devices would benefit the most from this, and streamline the testing and backporting process. 
And maybe something about getting in touch if you want to add a device type to this list?

Author: Added a simple note about additional device support planned. I think the other comments would be more applicable to the blog than the docs?
